components/kyma-operator/pkg/kymaoperation/steps/upgrade.go

```diff
@@ -0,0 +1,59 @@
+package steps
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+)
+
+type upgradeStep struct {
+	installStep
+	deployedRevision    int32
+	rollbackWaitTimeSec uint32
+}
+
+// Run method for upgradeStep triggers step upgrade via helm
+func (s upgradeStep) Run() error {
+
+	chartDir, err := s.sourceGetter.SrcDirFor(s.component)
+	if err != nil {
+		return err
+	}
+
+	releaseOverrides, releaseOverridesErr := s.overrideData.ForRelease(s.component.GetReleaseName())
+
+	if releaseOverridesErr != nil {
+		return releaseOverridesErr
+	}
+
+	upgradeResp, upgradeErr := s.helmClient.UpgradeRelease(
+		chartDir,
+		s.component.GetReleaseName(),
+		releaseOverrides)
+
+	if upgradeErr != nil {
+		upgradeErrMsg := fmt.Sprintf("Helm upgrade error: %s", upgradeErr.Error())
```

Comment:
Do we really need 3 separate errormsg vars? I think it would be cleaners to use one, and update the value

Reply:
I did not modify this code, just moved it. Nevertheless, I am not a big fan of reusing vars. It makes telling what a variable contains in a given point in the code almost impossible without careful inspection of every tiny bit of it.
I agree that it can be useful in some scenarios, though.
I will refactor this to use two vars - we need two anyway.